From cc843eb64f6599d9ede176f711fe6355015c30b1 Mon Sep 17 00:00:00 2001
From: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
Date: Mon, 8 Oct 2018 22:33:38 +0000
Subject: [PATCH] PictureLayerImpl: Use ceil() instead of ceilf()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

ceilf() is only formally mentioned in C++17, even though previous versions
of the standard say <cmath> should have the same contents as C's math.h.

libstdc++ does not ship ceilf() in its cmath, but the ceil(float) overload
that _is_ in the standard is available everywhere, so use that instead.

This fixes the build with libstdc++ after 77b67445ba ("Elements with fixed
bottom and top stick to top"):

    ../../cc/layers/picture_layer_impl.cc: In member function ‘void cc::PictureLayerImpl::UpdateViewportRectForTilePriorityInContentSpace()’:
    ../../cc/layers/picture_layer_impl.cc:738:16: error: ‘ceilf’ is not a member of ‘std’
               std::ceilf(-total_controls_height * hidden_ratio));  // bottom
                    ^~~~~
    ../../cc/layers/picture_layer_impl.cc:738:16: note: suggested alternative: ‘ceil’
               std::ceilf(-total_controls_height * hidden_ratio));  // bottom
                    ^~~~~
                    ceil

Bug: 819294
Cq-Include-Trybots: luci.chromium.try:android_optional_gpu_tests_rel;master.tryserver.blink:linux_trusty_blink_rel
Change-Id: Ida69eb3127ca132e7090c9aad0dd17dc314b6992
Reviewed-on: https://chromium-review.googlesource.com/c/1264537
Commit-Queue: David Bokan <bokan@chromium.org>
Reviewed-by: David Bokan <bokan@chromium.org>
Cr-Commit-Position: refs/heads/master@{#597709}
---
 cc/layers/picture_layer_impl.cc | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/cc/layers/picture_layer_impl.cc b/cc/layers/picture_layer_impl.cc
index 937255884a6e..020bcee55201 100644
--- a/cc/layers/picture_layer_impl.cc
+++ b/cc/layers/picture_layer_impl.cc
@@ -727,10 +727,10 @@ void PictureLayerImpl::UpdateViewportRectForTilePriorityInContentSpace() {
           1.f - layer_tree_impl()->CurrentBrowserControlsShownRatio();
 
       viewport_rect_for_tile_priority_in_content_space_.Inset(
-          0,                                                   // left
-          0,                                                   // top,
-          0,                                                   // right,
-          std::ceilf(-total_controls_height * hidden_ratio));  // bottom
+          0,                                                  // left
+          0,                                                  // top,
+          0,                                                  // right,
+          std::ceil(-total_controls_height * hidden_ratio));  // bottom
     }
   }
 }
-- 
2.19.1

